vajram/extensions/sql/vajram-sql/src/test/java/com/flipkart/krystal/vajram/sql/SQLReadTest.java

```diff
@@ -0,0 +1,197 @@
+class SQLReadTest {
+  private static final Logger logger = LoggerFactory.getLogger(SQLReadTest.class);
+  public static final int MAX_THREADS = Runtime.getRuntime().availableProcessors();
+
+  @SuppressWarnings("unchecked")
+  private static final Lease<SingleThreadExecutor>[] EXECUTOR_LEASES = new Lease[MAX_THREADS];
+
+  private static SingleThreadExecutorsPool EXEC_POOL;
+  private Injector injector;
+  private VajramKryonGraph graph;
+  private Lease<SingleThreadExecutor> executorLease;
+
+  @BeforeAll
+  static void beforeAll() throws LeaseUnavailableException {
+    EXEC_POOL = new SingleThreadExecutorsPool("Test", MAX_THREADS);
+    for (int i = 0; i < MAX_THREADS; i++) {
+      EXECUTOR_LEASES[i] = EXEC_POOL.lease();
+    }
+  }
+
+  @AfterAll
+  static void afterAll() {
+    stream(EXECUTOR_LEASES).forEach(Lease::close);
+  }
+
+
+  @BeforeEach
+  void setUp() {
+    this.executorLease = EXECUTOR_LEASES[0];
+    VajramKryonGraphBuilder builder = VajramKryonGraph.builder();
+    builder.loadFromPackage(SQLWrite.class.getPackageName());
+    this.injector = Guice.createInjector(new Utils.GuiceModule());
+    // Create the VajramKryonGraph
+    this.graph = builder.build();
+    graph.registerInputInjector(new VajramGuiceInputInjector(injector));
+
+  }
+
+  @Test
+  @Disabled("Requires MySQL server - enable manually when database is available")
+  void readfromMySQL_withMapResult() {
+    logger.info("Testing SELECT operation with Map result");
+    KrystexVajramExecutorConfig config =
+        KrystexVajramExecutorConfig.builder()
+            .kryonExecutorConfigBuilder(
+                KryonExecutorConfig.builder()
+                    .executorId("readFromMySQLTest")
+                    .executorService(executorLease.get()))
+            .build();
+    // SELECT query with parameters
+    String query = "select * from user_profile";
+    List<Object> parameters = Collections.emptyList();
+
+    // Execute the vajram without resultType (returns List<Map<String, Object>>)
+    CompletableFuture<SQLResult> future;
+    try (KrystexVajramExecutor vajramExecutor = graph.createExecutor(config)) {
+      future =
+          vajramExecutor.execute(
+              SQLRead_ReqImmutPojo._builder()
+                  .selectQuery(query)
+                  .parameters(parameters)
+                  // No resultType specified - defaults to Map
+                  ._build(),
+              KryonExecutionConfig.builder().executionId("test-execution-map-read").build());
+    }
+
+    // Assert results
+    System.out.println("Waiting for result...");
+    SQLResult result = future.join();
+    System.out.println("Received result.");
+    System.out.println("Queried rows: " + result.rows().size());
+
+    // Cast to Map type and print first row
+    if (!result.rows().isEmpty()) {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> firstRow = (Map<String, Object>) result.rows().get(0);
+      System.out.println("First row: " + firstRow);
```

Comment:
I would suggest to use mock Connection object and add assertions to all test cases.